api/coreservice_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package api
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	logfilter "github.com/iotexproject/iotex-core/api/logfilter"
+	"github.com/iotexproject/iotex-core/config"
+	"github.com/iotexproject/iotex-core/testutil"
+	"github.com/iotexproject/iotex-proto/golang/iotexapi"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+)
+
+func TestLogsInRange(t *testing.T) {
+	require := require.New(t)
+	svr, _, _, _, cleanCallback := setupTestServer(t)
+	defer cleanCallback()
+
+	testData := []struct {
+		filter      *filterObject
+		description string
+		logLen      int
+	}{
+		{
+			&filterObject{FromBlock: "1", ToBlock: "4"},
+			"blocks with four logs",
+			4,
+		},
+		{
+			&filterObject{FromBlock: "0x1"},
+			"default value for invalid string FromBlock & ToBlock",
+			4,
+		},
+		{
+			&filterObject{FromBlock: "5", ToBlock: "1"},
+			"invalid start and end height",
+			0,
+		},
+		{
+			&filterObject{FromBlock: "5", ToBlock: "5"},
+			"start block > tip height",
+			0,
+		},
+		{
+			// empty log
+			&filterObject{Address: []string{"0x8ce313ab12bf7aed8136ab36c623ff98c8eaad34"}, FromBlock: "1", ToBlock: "4"},
+			"empty log",
+			0,
+		},
+		{
+			&filterObject{FromBlock: "1", ToBlock: "4"},
+			"over 5000 pagenation size",
+			4,
+		},
+	}
+	for i, d := range testData {
+		t.Run(fmt.Sprintf("%v", d.description), func(t *testing.T) {
+			var filter iotexapi.LogsFilter
+			for _, ethAddr := range d.filter.Address {
+				ioAddr, err := ethAddrToIoAddr(ethAddr)
+				require.NoError(err)
+				filter.Address = append(filter.Address, ioAddr.String())
+			}
+			for _, tp := range d.filter.Topics {
+				var topic [][]byte
+				for _, str := range tp {
+					b, err := hexToBytes(str)
+					require.NoError(err)
+					topic = append(topic, b)
+				}
+				filter.Topics = append(filter.Topics, &iotexapi.Topics{Topic: topic})
+			}
+
+			// LogsInRange will assign default value if from, to returns error
+			from, err := strconv.ParseInt(d.filter.FromBlock, 10, 64)
+			if i == 1 {
+				require.Error(err)
+			} else {
+				require.NoError(err)
+			}
+			to, err := strconv.ParseInt(d.filter.ToBlock, 10, 64)
+			if i == 1 {
```

Comment:
It looks like for loop is unnecessary here because of the expression `i == 1` and `i == 5`.  `t.Run()` would be better for each test